File: app/src/main/java/com/devdream/cookall/core/api/APIRestClient.java
package com.devdream.cookall.core.api;

import com.devdream.cookall.core.utils.Constants;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public abstract class APIRestClient {

    private static Retrofit retrofit = null;

    public static Retrofit getClient() {
        if (retrofit == null) {
            retrofit = new Retrofit.Builder()
                    .baseUrl(Constants.API_BASE_URL)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();
        }
        return retrofit;
    }

}

File: app/src/main/java/com/devdream/cookall/core/api/APIRestClient.kt
package com.devdream.cookall.core.api

import com.devdream.cookall.core.utils.Constants

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object APIRestClient {

    private var retrofit: Retrofit? = null

    val client: Retrofit
        get() {
            if (null == retrofit) {
                retrofit = Retrofit.Builder()
                        .baseUrl(Constants.API.BASE_URL)
                        .addConverterFactory(GsonConverterFactory.create())
                        .build()
            }
            return retrofit
        }

}
